### PR TITLE
List + Solr work_key powered QueryCarousel

### PR DESCRIPTION
**Pull request by mekarpeles**
Allows List keys to be passed to `QueryCarousel` macro to power carousels

### Technical

This PR extends `QueryCarousel` so a list key may be provided, such as `/people/mekBot/lists/OL104041L/` and the when https://github.com/internetarchive/openlibrary/compare/master...list-querycarousel#diff-a6072af04af429689c35331c2a4a35fa2c23317099533cc2161ef6d8ba79df90R1034 is called, the list will be expanded into a set of work_keys which are *then* passed through to solr.

Note: The limit and offset are necessarily for paging through the List, but we **don't** want the limit and offset to be applied to solr because solr is going to receive a discrete list of work_ids (from the already paginated/offset List) and we want to return all of these works from solr. 

Will likely be unnecessary once/if Lists are in solr

### Testing

Adding a `{{ QueryParam("/people/mekBot/lists/OL104041L/") }}` (please check syntax) macro somewhere not too public (e.g. user profile page) should do the trick once this code is on testing. 
 
### Screenshot

Will add once it's on testing

### Stakeholders

@cdrini @seabelis

### CODE REVIEW

**Comment by cclauss** [comment]
Please run psf/black on this code to autofix the failing tests...
```
git diff "${BASE_BRANCH:-master}" -U0 | ./scripts/flake8-diff.sh
openlibrary/plugins/openlibrary/lists.py:210:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/openlibrary/lists.py:225:17: W503 line break before binary operator
openlibrary/plugins/openlibrary/lists.py:237:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/openlibrary/lists.py:251:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/openlibrary/lists.py:255:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/openlibrary/lists.py:269:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/openlibrary/lists.py:327:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/openlibrary/lists.py:367:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/worksearch/code.py:1011:1: E302 expected 2 blank lines, found 1
openlibrary/plugins/worksearch/code.py:1048:89: E501 line too long (98 > 88 characters)
openlibrary/plugins/worksearch/code.py:1109:89: E501 line too long (93 > 88 characters)
8     E302 expected 2 blank lines, found 1
2     E501 line too long (98 > 88 characters)
1     W503 line break before binary operator
```

File: openlibrary/plugins/openlibrary/lists.py [hunk 1]
```diff
@@ -323,31 +324,23 @@ class list_seed_yaml(list_seeds):
     encoding = "yml"
     content_type = 'text/yaml; charset="utf-8"'
 
-
-class list_editions(delegate.page):
```
Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/lists.py:327]: `list_editions` (not the json endpoint) appears to be dead code as there is no such template `type/list/editions.html`

`r"(/people/\w+/lists/OL\d+L)/editions"` i.e.
https://openlibrary.org/people/mekBot/lists/OL104041L/Popular_Books/editions?debug=true

File: openlibrary/macros/QueryCarousel.html [hunk 2]
```diff
@@ -19,7 +19,7 @@
   </form>
 
 $code:
-  params = { 'q': query }
+  params = { 'q': query, 'fields': '*,availability' }
```
Comment (mekarpeles) [inline openlibrary/macros/QueryCarousel.html:22]: Ensures the "load more" solr query behind the scene also fetches / shuffles in book availability for carousels

File: openlibrary/plugins/openlibrary/lists.py [hunk 3]
```diff
@@ -244,28 +235,33 @@ def get_list_data(self, lst):
...
-
     content_type = "application/json"
 
     def GET(self, key):
-        lst = web.ctx.site.get(key)
-        if not lst:
+        i = web.input()
+        raw = i.get("_raw") == "true"
+        lst = get_list(key, raw=raw)
+        if not lst or lst.type.key == '/type/delete':
```
Comment (jamesachamp) [inline openlibrary/plugins/openlibrary/lists.py:248]:
```suggestion
        if not lst or lst['type']['key'] == '/type/delete':
```
This is throwing an attribute error on testing.  Even with the above change, a KeyError is thrown when `raw` is false.  There should either be another check for `lst['links']['self'] here when `raw is false, or `type.key` should be added to the dictionary returned by `get_list()`.

**Review by jamesachamp — CHANGES_REQUESTED** [review]
The new ListCarousel macro seems to be working properly locally and on testing.  The list JSON endpoint is returning errors, though.

File: openlibrary/plugins/openlibrary/lists.py [hunk 4]
```diff
@@ -233,7 +222,9 @@ def get_list_data(self, lst):
                 "editions": lst.key + "/editions",
             },
             "name": lst.name or None,
-            "description": lst.description and six.text_type(lst.description) or None,
+            "description": (
```
Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/lists.py:225]:
```suggestion
            "type": { "key": lst.key },
            "description": (
```

File: openlibrary/plugins/openlibrary/lists.py [hunk 5]
```diff
@@ -233,7 +222,10 @@ def get_list_data(self, lst):
                 "editions": lst.key + "/editions",
             },
             "name": lst.name or None,
-            "description": lst.description and six.text_type(lst.description) or None,
+            "type": { "key": lst.key },
```
Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/lists.py:225]:
```suggestion
            "type": {"key": lst.key},
```

**Pull request merged**